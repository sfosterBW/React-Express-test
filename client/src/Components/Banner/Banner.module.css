.wrapper {
  margin-top: 0;
  margin-bottom: 3vmin;
}

.hero {
  background-color: #ccc;
  display: flex;
  box-sizing: border-box;
  align-items: center;
  justify-content: center;
}

.one {
  background-image: url("https://previews.123rf.com/images/berry2046/berry20461904/berry2046190400052/122038881-tropical-vector-realistic-green-leaves-seamless-pattern-pink-background-exotic-trendy-wallpaper.jpg");
}

.one .title {
  color: yellow;
  text-shadow: 0.8vmin 0.8vmin #ff0000;
}

.two {
  background-image: url("https://previews.123rf.com/images/darijashka/darijashka1904/darijashka190400034/123951263-summer-seamless-pattern-with-palm-leaves-and-tropical-flowers-hawaiian-motifs-perfect-for-wallpapers.jpg")
}

.two .title {
  color: yellow;
  text-shadow: 0.8vmin 0.8vmin #ff0000;
}

.three {
  background-image: url("https://previews.123rf.com/images/theerapol/theerapol1901/theerapol190100009/126453733-floral-seamless-tropical-pattern-botanical-wallpaper-illustration-in-hawaiian-style.jpg");
}

.three .title {
  color: yellow;
  text-shadow: 0.8vmin 0.8vmin #ff0000;
}

.small {
  padding: 4vmin 0;
  width: 70vw;
  border-radius: 0.5vmin;
}

.medium {
  padding: 6vmin 0;
  width: 70vw;
  border-radius: 0.5vmin;
}

.large {
  padding: 12vmin 0;
  width: 100vw;
}

.title {
  font-size: 8vmin;
  display: flex;
  box-sizing: border-box;
  margin: 0;
}
