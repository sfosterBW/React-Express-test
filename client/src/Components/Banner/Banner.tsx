import React, { FC } from 'react'
import styles from './Banner.module.css'

interface Props {
  theme?: 'one' | 'two' | 'three';
  size: 'small' | 'medium' | 'large';
  title: string;
}

const Banner: FC<Props> = ({ theme = 'one', size, title }) => {
  return (
    <section className={styles.wrapper} data-testid="wrapper">
      <div className={`${styles.hero} ${styles[size]} ${styles[theme]}`}>
        <h1 className={styles.title} data-testid="title">
          {title}
        </h1>
      </div>
    </section>
  )
}

export default Banner
