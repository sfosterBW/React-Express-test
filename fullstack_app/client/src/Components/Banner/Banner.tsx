import React, { FC } from 'react'
import styles from './Banner.module.css'

interface Props {
  flavour?: "flavourOne" | "flavourTwo" | "flavourThree"
  size: "small" | "medium" | "large"
  title: string
}

const Banner: FC<Props> = ({flavour = "flavourOne", size, title}) => {
  return (
    <section className={styles.wrapper}>
      <div className={`${styles.hero} ${styles[size]} ${styles[flavour]}`}>
        <h1 className={styles.title}>{title}</h1>
      </div>
    </section>
  )
}

export default Banner
