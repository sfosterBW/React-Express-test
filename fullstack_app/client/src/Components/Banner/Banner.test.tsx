import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import Banner from './Banner'

describe('the Alert component', () => {
  const size = "large"
  const title = "Title"
  const component = <Banner size={size} title={title}/>
  const wrapper = mount(component)

  it('renders with the right structure', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper.find('section')).toHaveLength(1)
    expect(wrapper.find('div')).toHaveLength(1)
    expect(wrapper.find('h1')).toHaveLength(1)
    expect(wrapper.find('h1').text()).toEqual(title)
  })

  it('renders the same as last time with props', () => {
    const tree = renderer
      .create(component)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
