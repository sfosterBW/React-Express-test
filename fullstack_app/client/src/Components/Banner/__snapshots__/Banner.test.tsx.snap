// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Alert component renders the same as last time with props 1`] = `
<section
  className="wrapper"
>
  <div
    className="hero large flavourOne"
  >
    <h1
      className="title"
    >
      Title
    </h1>
  </div>
</section>
`;
